packages/dds/tree/src/test/shared-tree/treeCheckout.spec.ts

```diff
@@ -637,6 +638,94 @@ describe("sharedTreeView", () => {
 		checkout1Revertibles.unsubscribe();
 		checkout2Revertibles.unsubscribe();
 	});
+
+	describe("branches with schema edits can be rebased", () => {
+		it("over non-schema changes", () => {
+			const provider = new TestTreeProviderLite(1);
+			const checkout1 = provider.trees[0].checkout;
+
+			checkout1.updateSchema(intoStoredSchema(jsonSequenceRootSchema));
+			checkout1.editor.sequenceField(rootField).insert(
+				0,
+				cursorForJsonableTreeField([
+					{ type: leaf.string.name, value: "A" },
+					{ type: leaf.string.name, value: "B" },
+					{ type: leaf.string.name, value: "C" },
+				]),
+			);
+
+			const branch = checkout1.fork();
+
+			// Remove "A" on the parent branch
+			checkout1.editor.sequenceField(rootField).remove(0, 1);
+
+			// Remove "B" on the child branch
+			branch.editor.sequenceField(rootField).remove(1, 1);
+			branch.updateSchema(intoStoredSchema(stringSequenceRootSchema));
+			// Remove "C" on the child branch
+			branch.editor.sequenceField(rootField).remove(1, 1);
+			validateTreeContent(branch, {
+				schema: stringSequenceRootSchema,
+				initialTree: ["A"],
+			});
+
+			branch.rebaseOnto(checkout1);
+
+			// The schema change and any changes after that should be dropped,
+			// but the changes before the schema change should be preserved
+			validateTreeContent(branch, {
+				schema: jsonSequenceRootSchema,
+				initialTree: ["C"],
+			});
+		});
+
+		// AB#7256: This test fails because purging repair data upon application of schema changes makes it impossible
+		// to roll back the changes that were before that schema change on the branch being rebased.
```

Comment:
Work item mentions:

> This bug leads to an unrecoverable error. In practice the user could just reload the app which would be successful, but the user would lose their local changes.

Is that actually true in a collaborative setting rather than branch setting? That's a bit surprising to me. Seems like a readonly client (A) 'observing' two clients (B, C) concurrently changing schema might involve similar rebases to keep A's view of B & C's branches up-to-date. Or is this because things only go wrong in Forest and A never attempts to materialize a forest from the perspective of B/C?